### PR TITLE
Avoid use of sprintf() in learnt_info().  That's part of resolving …

### PR DESCRIPTION
**Pull request by backwardsEric**
… https://github.com/hengband/hengband/issues/2858 and https://github.com/hengband/hengband/issues/2859 .

### CODE REVIEW

File: src/blue-magic/learnt-info.cpp [hunk 1]
```diff
@@ -37,20 +38,19 @@ static void set_bluemage_damage(PlayerType *player_ptr, MonsterAbilityType ms_ty
...
  * @param power モンスター魔法のID
+ * @return std::string パワーについてのコメント
  */
-void learnt_info(PlayerType *player_ptr, char *p, MonsterAbilityType power)
+std::string learnt_info(PlayerType *player_ptr, MonsterAbilityType power)
 {
     PLAYER_LEVEL plev = get_pseudo_monstetr_level(player_ptr);
-
-    strcpy(p, "");
+    std::string comment;
```
Comment (Hourier) [inline src/blue-magic/learnt-info.cpp:53]: This variable isn't needed.
I comment the reason later.

File: src/blue-magic/learnt-info.cpp [hunk 2]
```diff
@@ -92,7 +92,7 @@ void learnt_info(PlayerType *player_ptr, char *p, MonsterAbilityType power)
     case MonsterAbilityType::BA_MANA:
     case MonsterAbilityType::BA_DARK:
     case MonsterAbilityType::BA_LITE:
-        set_bluemage_damage(player_ptr, power, plev, KWD_DAM, p);
+        comment = set_bluemage_damage(player_ptr, power, plev, KWD_DAM);
```
Comment (Hourier) [inline src/blue-magic/learnt-info.cpp:95]: This sentences can be changed:

Before:
```
comment = set_bluemage_damage(player_ptr, power, plev, KWD_DAM);
break;
```

After:
```
return set_bluemage_damage(player_ptr, power, plev, KWD_DAM);
```

**Review by Hourier — COMMENTED** [review]
Thank you for changing char\[\] to std::string!
I added some comments.
Would you please confirm and fix them?

File: src/blue-magic/learnt-info.cpp (hunk 2 above)
Comment (backwardsEric) [reply src/blue-magic/learnt-info.cpp]: The forced push changed the case statement in learnt_info() so each case directly returns a result.

File: src/blue-magic/learnt-info.cpp [hunk 3]
```diff
@@ -88,14 +87,13 @@ void learnt_info(PlayerType *player_ptr, char *p, MonsterAbilityType power)
     case MonsterAbilityType::S_HI_DRAGON:
     case MonsterAbilityType::S_AMBERITES:
     case MonsterAbilityType::S_UNIQUE:
-        break;
+        return std::string();
```
Comment (Hourier) [inline src/blue-magic/learnt-info.cpp:90]: You can delete from SHRIEK to S\_UNIQUE and DISPEL.
(They go to default)

**Review by Hourier — COMMENTED** [review]
Thank you for reflecting my comments.
I confirmed them and found 1 more comment.
Please change it.

**Comment by backwardsEric** [comment]
The last forced push removes the cases from learnt_info() which have the same handling as the default case.

**Pull request merged**